network.go

```diff
@@ -438,11 +438,9 @@ func (n *network) updateSvcRecord(ep *endpoint, isAdd bool) {
 	var epList []*endpoint
 	n.WalkEndpoints(func(e Endpoint) bool {
 		cEp := e.(*endpoint)
-		cEp.Lock()
-		if cEp.container != nil {
+		if _, hasSandbox := cEp.getSandbox(); hasSandbox {
```

Comment:
This is the reason you might have still maintained the endpoint methods `addHostEntries/deleteHostEntries`. But those methods should really become `Sandbox` methods and here you should generate a list of Sandboxes belonging to the network and then iterate to invoke them